### PR TITLE
Code Review for <fetch-api-data> at Math Magicians project.

### PR DESCRIPTION
**Pull request by luigirazum**
# Hello Dear Code Reviewers Team 👋,
### The following requirements 📌 were fulfilled this time:
- 🎯 <b>General requirements</b>
    - 📌 Make sure that there are no linter errors.
        - ✅ All linters are green 🟢.
    - 📌 Make sure that you used correct git-flow.
        - ✅ `fetch-api-data` feature branch was created to develop the feature and follow the `git-flow`.
    - 📌 Make sure that you documented your work in a professional way.
        - ✅ The project has a README file with all its information.

- 🎯 <b>Project requirements</b>
    - 📌 Use the following external API: API Ninjas.
        - ✅ `https://api-ninjas.com/api/quotes` was used
    - 📌 Create a new component for displaying a quote.
        - ✅ `Quote` component was created.
    - 📌 Fetch data inside of a new component.
        - ✅ A quote from the API is retrieved and inserted into the `Quote` component.
    - 📌 Show a <kbd>loading state</kbd>.
        - ✅ When the app is fetching the quote from the API shows a <kbd>loading...</kbd> message is shown.
    - 📌 Show an <kbd>error state</kbd>.
        - ✅ When an error occurs during fetching the quote from the API an <kbd>error has ocurred</kbd> message is shown.
    - 📌 Import the new component into your main component (next to the `Calculator`).
        - ✅ `Quote` component was imported in main `App` component and render next to the `Calculator`.
        - 👉 It should be on the same page as the `Calculator`.
            - ✔ `Quote` component is render in the same page with `Calculator`.
            working-api

### Luis 🙋‍♂️

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,11 +1,61 @@
...
+        setHasError(false);
+      } catch (error) {
+        setHasError(true);
+      }
+
+      setIsLoading(false);
+    };
+
+    fetchQuote();
+  }, []);
```
Comment (amejid) [inline src/App.js:42]: - [ ] Great job handling errors very well. :+1: Let's do a bit of refactoring to achieve the project requirements. All these data fetching and error handling belongs to the `Quote` component, we are not using any of this data in the `App` component itself or any of its children other than `Quote`. Kindly move this logic to the `Quote` component. 🤝️

**Project requirements**
Screenshot from 2023-03-23 05-09-11

File: src/App.js [hunk 2]
```diff
@@ -1,11 +1,61 @@
...
+      <Calculator />
+      { isLoading
+        ? (<p>Daily Quote is Loading...</p>)
+        : (
+          <Quote
+            quote={dailyQuote.quote}
+            author={dailyQuote.author}
+          />
+        ) }
+      { hasError && <p>Something went wrong while fetching the daily quote.</p> }
```
Comment (amejid) [inline src/App.js:56]: - [ ] Please do the same thing for these pieces of code too. After completing this you'll have a structure like the following.  🧐️

```
<>
  <h1>Welcome to Math Magicians</h1>
  <Calculator />
  <Quote />
</>
```

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @luigirazum, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- Correct use of Gitflow :heavy_check_mark: 
- Professional README :heavy_check_mark: 
- Data fetching working as expected :heavy_check_mark: 
- Clean code :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 1 above)
Comment (luigirazum) [reply src/App.js]: Thanks for your feedback @amejid.
All Quote logic was move from App into Quote component.

File: src/App.js (hunk 2 above)
Comment (luigirazum) [reply src/App.js]: Thanks again @amejid.
App component renders Calculator and Quote components.

**Review by m05e5 — APPROVED** [review]
**Hi @luigirazum 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Wll done implementing the requested changes
✔️ Clean and organized code 
✔️ no linters error

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Comment by luigirazum** [comment]
Thanks @m05e5 👍

**Pull request merged**